pkg/common-controller/snapshot_controller.go

```diff
@@ -425,27 +460,47 @@ func (ctrl *csiSnapshotCommonController) syncUnreadySnapshot(snapshot *crdv1.Vol
 			// update snapshot status failed
 			klog.V(4).Infof("failed to update snapshot %s status: %v", utils.SnapshotKey(snapshot), err)
 			ctrl.updateSnapshotErrorStatusWithEvent(snapshot, v1.EventTypeWarning, "SnapshotStatusUpdateFailed", fmt.Sprintf("Snapshot status update failed, %v", err))
+			snapshotStatusType = metrics.SnapshotStatusTypeControllerError
 			return err
 		}
+
+		snapshotStatusType = metrics.SnapshotStatusTypeSuccess
```

Comment:
at this point, the VolumeSnapshot may or may not have "ready" set to true yet. This simply means the controller has successfully bound the volume snapshot to volume snapshot content. Sidecar will call CSI "ListSnapshots" to update  content status, and then common controller will resync in next reconciliation. setting status to success here effectively ends the e2e latency reporting which is wrong IMO.